msteams-platform/toolkit/debug-teams-app-visual-studio.md

```diff
@@ -0,0 +1,142 @@
+
+## Prerequisites
+
+* Install the latest version of Teams Toolkit for Visual Studio.
+* To run / debug a Bot or Message Extension app, you'll need to set up [Ngrok](https://ngrok.com/) first.
+Ngrok is used to forward external messages from Azure Bot Framework to your local machine.
+
+## Debug your app locally
+
+### Set up Ngrok (Only for Bot and Message Extension app)
+
+Use a Command Prompt to run this command:
+
+```
+ngrok http 5130.
+```
+
+### Set up your Teams Toolkit
+
+Perform the following steps to debug your app after you create a new app using the Teams Toolkit:
+
+1. In Visual Studio Solution Explorer, right click on your project file and select "Teams Toolkit"
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-selectTeamsToolkit.png" alt-text="localdebug selectTeamsToolkit":::
+
+Then select "Prepare Teams app dependencies".
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-selectTeamsPrepare.png" alt-text="localdebug selectTeamsPrepare":::
+
+You'll be asked to sign in to your Microsoft 365 account.
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-selectAccountContinue.png" alt-text="localdebug selectAccountContinue":::
+
+   > [!Note]
+   > Your M365 account need to have the sideloading permission to ensure Teams app can be uploaded to your tenant, otherwise you will end up with failure to see your Teams app running in Teams client. Learn more about sideloading permission by visiting <https://aka.ms/teamsfx-sideloading-option>.
+
+2. Go to the "Debug" menu > select "Start Debugging" or directly press **F5**. Visual Studio will launch the Teams app inside Microsoft Teams client in a browser.
+
+:::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-menu-selectDebug.png" alt-text="localdebug menu selectDebug":::
+
+:::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-menu-startDebugging.png" alt-text="localdebug menu startDebug":::
+
+   > [!Note]
+   > Learn more by visiting <https://aka.ms/teamsfx-vs-debug>.
+
+3. Once Microsoft Teams is loaded, select the "Add" button to install your app in Teams.
+
+    :::image type="content" source="../assets/images/teams-toolkit-v2/debug-VS/VS-localdebug-addApp.png" alt-text="localdebug addApp":::
+
+   > [!TIP]
+   > You can use hotreload function of VS during debug. Learn more by visiting <https://aka.ms/teamsfx-vs-hotreload>.
+
+4. If you're debugging Notification Bot app, and if you selected http trigger when creating the project. You need to post HTTP request to <http://localhost:5130/api/notification> to trigger notification. You can use any API tool: curl (Windows Command Prompt), Postman, etc.
```

Comment:
comma

Reply:
Please clarify the comment. Unable to identify the change suggested.

Reply:
If you're debugging Notification Bot app, and if you selected http trigger when creating the project**,** you need to post HTTP request to <http://localhost:5130/api/notification> to trigger notification. You can use any API tool: curl (Windows Command Prompt), Postman, etc.

Reply:
This section has been fixed.